annotation/tests/test_jobs.py

```diff
@@ -497,3 +505,354 @@ def test_update_user_overall_load(tasks: Tuple[ManualAnnotationTask]):
+    mock_session.query().filter_by().first.return_value = None
+    with pytest.raises(services.WrongJobError):
+        services.get_job(mock_session, 1, "tenant")
+
+
+def test_get_job_success():
+    mock_session = MagicMock()
+    mock_session.query().filter_by().first.return_value = Job(
+        job_id=1, tenant="tenant"
+    )
+    result = services.get_job(mock_session, 1, "tenant")
+    expected_result = Job(job_id=1, tenant="tenant")
+    assert result == expected_result
+
+
+def test_update_files(files: Tuple[File, ...]):
+    mock_session = MagicMock()
+    mock_session.query().filter().with_for_update().all.return_value = [
+        files[0],
+        files[1],
+    ]
+    tasks = [
+        {"file_id": 1, "is_validation": True, "pages": [1, 2, 3]},
+        {"file_id": 2, "is_validation": False, "pages": [4, 5, 6]},
+    ]
+    services.update_files(mock_session, tasks, 1)
+    assert files[0].distributed_validating_pages == [1, 2, 3]
+    assert files[1].distributed_annotating_pages == [4, 5, 6]
+
+
+def test_delete_tasks(
+    tasks: Tuple[ManualAnnotationTask, ...], files: Tuple[File, ...]
+):
+    mock_session = MagicMock()
+    mock_session.query().filter().with_for_update().first.return_value = files[
+        0
+    ]
+    expected_delete_calls = [
+        unittest.mock.call(tasks[0]),
+        unittest.mock.call(tasks[5]),
+    ]
+
+    expected_recalculate_calls = files[0]
+    expected_user_load_calls = [
+        unittest.mock.call(mock_session, tasks[0].user_id),
+        unittest.mock.call(mock_session, tasks[5].user_id),
+    ]
+    with patch(
+        "annotation.jobs.services.recalculate_file_pages"
+    ) as mock_recalculate_file_pages, patch(
+        "annotation.jobs.services.update_user_overall_load"
+    ) as mock_update_user_overall_load:
+        services.delete_tasks(mock_session, {tasks[0], tasks[5]})
+        mock_session.delete.assert_has_calls(
+            expected_delete_calls, any_order=True
+        )
+        mock_recalculate_file_pages.assert_called_with(
+            mock_session, expected_recalculate_calls
+        )
+        mock_update_user_overall_load.assert_has_calls(
+            expected_user_load_calls, any_order=True
+        )
+
+
+@pytest.mark.parametrize(
+    (
+        "new_validators",
+        "new_annotators",
+        "job_validators",
+        "job_annotators",
+        "expected_annotators",
+        "expected_validators",
+    ),
+    (
+        (
+            {UUID(int=1), UUID(int=2)},
+            {UUID(int=3)},
+            (User(user_id=UUID(int=1)), User(user_id=UUID(int=2))),
+            (User(user_id=UUID(int=3)), User(user_id=UUID(int=4))),
+            {UUID(int=4)},
+            set(),
+        ),
+        (
+            {UUID(int=3)},
+            {UUID(int=2), UUID(int=1)},
+            (User(user_id=UUID(int=4)), User(user_id=UUID(int=3))),
+            (User(user_id=UUID(int=2)), User(user_id=UUID(int=1))),
+            set(),
+            {UUID(int=4)},
```

Comment:
let's make it set of ints (ids) for UUIDs. You can build the full data type in the test.